From 841b358a552ac5706855fd6be2cb2efb7a5c2673 Mon Sep 17 00:00:00 2001
From: Romain Naour <romain.naour@openwide.fr>
Date: Sat, 11 Apr 2015 12:46:05 +0200
Subject: [PATCH] autotools: add --disable-libeeze

This allows to disable libeeze if necessary and remove the
dependency on libudev.

When cross-compiling, we want to build only edje_cc and eet
binaries for the host before starting the build for the
target.

In normal case it's not recommended hence a warning.

Signed-off-by: Romain Naour <romain.naour@openwide.fr>
---
 configure.ac | 34 +++++++++++++++++++++++++++++++++-
 1 file changed, 33 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 604c0fb..7ffafc3 100644
--- a/configure.ac
+++ b/configure.ac
@@ -634,6 +634,31 @@ AC_ARG_ENABLE([gui],
    ])
 AM_CONDITIONAL([BUILD_GUI], [test "x${build_gui}" = "xyes"])
 
+want_libeeze="yes"
+AC_ARG_ENABLE([libeeze],
+   [AS_HELP_STRING([--disable-libeeze],[disable Eeze device library @<:@default=enable@:>@])],
+   [
+    if test "x${enableval}" = "xyes"; then
+       want_libeeze="yes"
+    else
+       want_libeeze="no"
+       CFOPT_WARNING="yes"
+    fi
+   ])
+
+build_libeeze="yes"
+if test "x${want_libeeze}" = "xyes" ; then
+    if test "x${have_linux}" = "xyes" ; then
+        build_libeeze="yes"
+    else
+        build_libeeze="no"
+        want_libeeze="no"
+        CFOPT_WARNING="yes"
+    fi
+else
+    build_libeeze="no"
+fi
+
 AC_ARG_ENABLE([systemd],
    [AS_HELP_STRING([--enable-systemd],[Enable systemd support. @<:@default=disabled@:>@])],
    [
@@ -2937,7 +2962,7 @@ have_libmount_new="no"
 have_libmount_old="no"
 have_eeze_mount="no"
 
-EFL_LIB_START_OPTIONAL([Eeze], [test "${have_linux}" = "yes"])
+EFL_LIB_START_OPTIONAL([Eeze], [test "x${build_libeeze}" = "xyes"])
 
 ### Additional options to configure
 AC_ARG_WITH([mount],
@@ -5000,6 +5025,13 @@ if test -n "$CFOPT_WARNING"; then
     echo "to both as it will break your build if set to eo."
     echo "_____________________________________________________________________"
   fi
+  if test "x${want_libeeze}" = "xno"; then
+    echo "_____________________________________________________________________"
+    echo "Libeeze has been disabled, and it is used heavily for support of"
+    echo "removable devices etc. and disabling this will hurt support for"
+    echo "Enlightenment and its filemanager."
+    echo "_____________________________________________________________________"
+  fi
   echo "_____________________________________________________________________"
   echo ""
   echo "==-- WARNING --=="
-- 
1.9.3

